tests/test_validation.py

```diff
@@ -208,3 +206,296 @@ def test_validation_on_write(tmp_path):
 
     with pytest.warns(ValidationWarning):
         model.save(file_path)
+
+def test_validate_on_assignment_default():
+    model = ValidationModel()
+    assert model._validate_on_assignment==True
+
+    with pytest.warns(ValidationWarning):
+        model.meta.string_attribute = 42  # Bad assignment
+    assert model.meta.string_attribute is None
+
+
+def test_validate_on_assignment_false(tmp_path):
```

Comment:
It would be good to combine this with the previous test and parametrize.  This true for a number of tests below too.